pyro/contrib/bnn/hidden_layer.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, division, print_function
+
+
+import torch
+from torch.distributions.utils import lazy_property
+import torch.nn.functional as F
+
+from pyro.contrib.bnn.utils import adjoin_ones_vector
+from pyro.distributions.torch_distribution import TorchDistribution
+
+
+class HiddenLayer(TorchDistribution):
```

Comment:
Should this also inherit from `torch.nn.Module`?

Reply:
maybe, i am always wary of multiple inheritance because i'm not always sure how it works

Reply:
although probably not. since X changes from mini-batch to mini-batch in a typical use case you remake these on the fly as you go. so we don't really need any of the stateful enhancements that `nn.Module` bbrings